tests/reconfiguration.py

```diff
@@ -610,6 +611,170 @@ def run_join_old_snapshot(args):
                 pass
 
 
+def get_current_config(network):
```

Comment:
Could we use the new https://microsoft.github.io/CCF/main/operations/operator_rpc_api.html#get--service-configuration endpoint for this? 

Also, I think the `service-configuration` URI doesn't follow the existing convention we have for naming endpoints. I think `service/configuration` would be better.

Reply:
Ah wait, this is about network configuration. I think my comment about using the dedicated endpoint still holds though.

Reply:
Yes, this is the network config. At this point I want the list of nodes that's currently recorded on the ledger for later comparison. I don't want to test what any of the nodes think the config is, I want to test that things are recorded correctly on the ledger. 

(Sure, name change is fine)

Reply:
Because the latest state of the ledger (`get_latest_ledger_public_state()`) contains the same data than the KV store of the node, asking the node directly or reading the ledger should be equivalent. And as we found out the hard way, reading the ledger from the end-to-end tests can be flaky, it is preferable to send an RPC to the node to do so. In this case, AIUI, the `GET /node/consensus` should return the same value than what you're reading from the ledger?

Reply:
Yeah, I looked into that a little earlier today too. It basically gets the ledger from the primary anyways, so no big difference. Perhaps I should look into reading/auditing the ledger files after or at the end of the test to make sure it's all where it should be.

Reply:
I feel we should really do that validation on the ledger files after the test finishes. We can leave that for a separate PR, but is `read_ledger.py` as in the governance history test, the right way to access that?

Reply:
Yes, reading the ledger once all the nodes are stopped is safer/better. You should be able to iterate over the ledger without `read_ledger.py`, by directly using the `ccf.ledger.Ledger` object (see https://microsoft.github.io/CCF/main/audit/python_library.html#tutorial).